src/lib/DatePicker.svelte

```diff
@@ -64,37 +77,38 @@
     }
 
     const maxDate = getMonthLength(newYear, newMonth)
-    const newDate = Math.min(value.getDate(), maxDate)
-    setValue(
+    const newDate = Math.min(tmpPickerDate.getDate(), maxDate)
+    setPickerDate(
       new Date(
         newYear,
         newMonth,
         newDate,
-        value.getHours(),
-        value.getMinutes(),
-        value.getSeconds(),
-        value.getMilliseconds()
+        tmpPickerDate.getHours(),
+        tmpPickerDate.getMinutes(),
+        tmpPickerDate.getSeconds(),
+        tmpPickerDate.getMilliseconds()
       )
     )
   }
-  $: getMonth(value)
+  $: getMonth(tmpPickerDate)
   $: setMonth(month)
 
-  let dayOfMonth = value.getDate()
-  $: dayOfMonth = value.getDate()
+  let dayOfMonth = tmpPickerDate.getDate();
+  $: dayOfMonth = tmpPickerDate.getDate();
```

Comment:
I think we might want `dayOfMonth` to be `null` when `value` is `null`

Reply:
I don't think it can ever be null currently because there is always a default date.

Reply:
`tmpPickerDate` can't, but `value` can be null. I was imagining that when `DateInput` is null, `DatePicker` should be null and as a result not have any date highlighted.

Right now, `DatePicker` starts as `null` but the `defaultDate`'s `dayOfMonth` is highlighted, so to a user it looks like that's already selected.

Reply:
IMHO - until they click on a specific day the value should stay unassigned - otherwise as soon as they launch the picker the date will get assigned to the default without the user actually selecting a day as a side effect.

Reply:
Updating `dayOfMonth` doesn't update the `value`, so that wouldn't happen. If `dayOfMonth` is null, we would make sure no date shows up as selected before the user actually selects a date

Reply:
> Right now, DatePicker starts as null but the defaultDate's dayOfMonth is highlighted, so to a user it looks like that's already selected.

^-- Agree that this is still an issue. What do you think about having the picker show the default date in a lighter highlighted style - similar to the placeholder? Then when they select it it changes style to what you have now or something that clearly indicates the date is selected?

Reply:
I don't think anything should be highlighted by default, really

Reply:
...so then the default date would effectively be a default month and year to start the picker with no day selected - i don't think it's being used for anything else at this point. 

Not sure about the correct UX for this, but I searched for "w3c" and "date picker" and the first example shows the date highlighted in the picker before it's selected: 
https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html

Reply:
I think it's pretty common to not have a date highlighted when no date is selected. [Blueprint](https://blueprintjs.com/docs/#datetime/dateinput) and [Airbnb](https://www.airbnb.com/) both do it.

If `DatePicker` uses a default value anyway, I don't think it would make sense to allow it to be `null`

Reply:
Implemented on my `nullable` branch